rs/execution_environment/src/hypervisor.rs

```diff
@@ -438,12 +442,23 @@ impl Hypervisor {
             ),
         }
         let caller = api_type.caller();
+        let subnet_available_callbacks = round_limits.subnet_available_callbacks.max(0) as u64;
+        let canister_available_callbacks =
+            system_state
+                .call_context_manager()
```

Comment:
> If there's no `call_context_manager`, can we really have any available callbacks?

No, but then we wouldn't end up here (executing a message). The alternative would be to `unwrap()`, but that seems fraught (in more ways than one). Added a comment to clarify this.

> I.e., if we could have something like `Option<CallContextManagerLimits>` in `RoundLimits` and the `CallContextManager::apply_limits`.

Technically we could, but this `apply_limits()` would have to accept both `subnet_available_callbacks` (because it's a subnet-wide measurement) and `canister_guaranteed_callback_quota` (because it's a configuration value and we can't persist it in every `CallContextManager`across the replicated state). So IMHO instead of it keeping all the logic inside `CallContextManager`, it would be spreading it out across `Hypervisor` and `CallContextManager`.

That being said, if you feel strongly about this, it can definitely be done.